src/seraphis_mocks/mock_http_client_pool.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2022, The Monero Project
```

Comment:
These http_client_pool classes can probably go in the src/net folder, yeah?

Reply:
Agree, I'll throw them in there once it's in a ready-for-use state